File: 009_own_funcs/program.nim

#import std/[math, strutils]
import std/math


proc pow2(base: int64, n: int64): int64 =
  var p: int64 = 1
  
  for i in 1'i64 .. n :
    p = p * base
  result = p


for i in 0'i64 .. 10 :
  echo i, ":"
  echo pow2(2, i)
  echo pow2(-3, i)
  echo pow2(3, i)
  echo pow2(4, i)
  echo pow2(-4, i)
  echo pow2(5, i)
  echo "---"

echo "done!"

File: 009_own_funcs/program.c

#include <stdio.h>


//int pow2(long long m, long long n);

int pow2(long long base, long long n){
  long long i, p;
  p = 1;
  
  for(i = 1; i <= n; i++){
    p = p * base;
  }
  return p;
}


int main(void) {
  long long i;
  
  for(i = 0; i <= 10; i++){
    printf("%lld:\n%d\n%d\n%d\n%d\n%d\n%d\n---\n", i,
      pow2(2, i), pow2(-3, i),
      pow2(3, i), pow2(4, i),
      pow2(-4, i), pow2(5, i) );
  }
  
  printf("done!\n");
  
  return 0;
}


